src/NServiceBus.Testing/TestableMessageHandlerContext.cs

```diff
@@ -15,15 +16,15 @@ public TestableMessageHandlerContext(IMessageCreator messageCreator)
             this.messageCreator = messageCreator;
         }
 
-        public IList<TimeoutMessage<object>> TimeoutMessages { get; } = new List<TimeoutMessage<object>>();
+        public IList<SentMessage<object>> SentMessages => sentMessages.ToArray();
```

Comment:
I wanted to have it in a collection type which allows assertions about message order. What do you think about changing it to return arrays directly?

I wanted to avoid to expose concurrent collections to users, since at assertion time, concurrency should no longer be relevant.